pv_site_api/enode_auth.py

```diff
@@ -0,0 +1,37 @@
+import os
+
+import httpx
+
+
+class EnodeAuth(httpx.Auth):
+    def __init__(self, access_token):
+        self.access_token = access_token
+
+    def auth_flow(self, request):
+        # Add the Authorization header to the request using the current access token
+        request.headers["Authorization"] = f"Bearer {self.access_token}"
+        response = yield request
+
+        if response.status_code == 404:
+            # The access token is no longer valid, refresh it
+            token_response = yield self._build_refresh_request()
+            self._update_access_token(token_response)
+            # Update the request's Authorization header with the new access token
+            request.headers["Authorization"] = f"Bearer {self.access_token}"
+            # Resend the request with the new access token
+            response = yield request
+        return response
+
+    def _build_refresh_request(self):
+        # TODO: Add environment variables
+        client_id = os.getenv("CLIENT_ID")
+        client_secret = os.getenv("CLIENT_SECRET")
+        basic_auth = httpx.BasicAuth(client_id, client_secret)
+        url = os.getenv("ENODE_TOKEN_URL")
+        data = {"grant_type": "client_credentials"}
+        request = next(basic_auth.auth_flow(httpx.Request("POST", url, data=data)))
+        return request
```

Comment:
Remove comment and add some spacing for clarity.
```suggestion
        client_id = os.getenv("CLIENT_ID")
        client_secret = os.getenv("CLIENT_SECRET")
        basic_auth = httpx.BasicAuth(client_id, client_secret)

        url = os.getenv("ENODE_TOKEN_URL")
        data = {"grant_type": "client_credentials"}
        flow = basic_auth.auth_flow(httpx.Request("POST", url, data=data))
        request = next(flow)
        return request
```